File: codegen/results/v001/qwen3-coder_30b/20260409_103926/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

// Test cases
const testCases = [
    { price: 100, quantity: 5, description: "quantity 5 (less than 10)" },
    { price: 100, quantity: 9, description: "quantity 9 (boundary condition)" },
    { price: 100, quantity: 10, description: "quantity 10 (boundary condition)" },
    { price: 100, quantity: 49, description: "quantity 49 (boundary condition)" },
    { price: 100, quantity: 50, description: "quantity 50 (boundary condition)" },
    { price: 50, quantity: 25, description: "quantity 25 (between 10 and 49)" },
    { price: 20, quantity: 75, description: "quantity 75 (greater than 50)" }
];

let passedTests = 0;
let totalTests = testCases.length;

for (const testCase of testCases) {
    const result = calculateDiscount(testCase.price, testCase.quantity);
    let expected: number;
    
    if (testCase.quantity < 10) {
        expected = testCase.price * testCase.quantity;
    } else if (testCase.quantity < 50) {
        expected = testCase.price * testCase.quantity * 0.9;
    } else {
        expected = testCase.price * testCase.quantity * 0.75;
    }
    
    if (result === expected) {
        console.log(`PASS: ${testCase.description}`);
        passedTests++;
    } else {
        console.log(`FAIL: ${testCase.description} (expected: ${expected}, got: ${result})`);
    }
}

console.log(`\nTest Results: ${passedTests}/${totalTests} tests passed`);

File: codegen/results/v001/qwen3-coder_30b/20260409_103926/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75

# Test cases
test_cases = [
    (10.0, 5, "price 10.0, quantity 5 (no discount)"),
    (10.0, 10, "price 10.0, quantity 10 (10% discount)"),
    (10.0, 49, "price 10.0, quantity 49 (10% discount)"),
    (10.0, 50, "price 10.0, quantity 50 (25% discount)"),
    (5.0, 9, "price 5.0, quantity 9 (no discount)"),
    (15.0, 25, "price 15.0, quantity 25 (10% discount)"),
    (8.0, 75, "price 8.0, quantity 75 (25% discount)")
]

# Run tests
for price, quantity, description in test_cases:
    result = calculate_discount(price, quantity)
    
    # Calculate expected result
    if quantity < 10:
        expected = price * quantity
    elif quantity < 50:
        expected = price * quantity * 0.9
    else:
        expected = price * quantity * 0.75
    
    if result == expected:
        print(f"PASS: {description}")
    else:
        print(f"FAIL: {description} (expected: {expected}, got: {result})")